src/pages/AllDao.vue

```diff
@@ -1,110 +1,282 @@
 import DaoStore from "../store/DaoID";
 import { executeReq, ALL_DAOS_REQ } from "@/api";
-import { DaoItemType } from "@/types";
+import { DaoTableData } from "@/types";
+import WalletStore from "@/store/WalletStore";
+import { SearchOutlined } from "@ant-design/icons-vue";
 
 export default defineComponent({
 	name: "AllDao",
+	components: {
+		SearchOutlined,
+	},
 	data() {
 		return {
-			error: "",
 			key: "DaoKey",
+			columns: [
+				{
+					title: "Dao App ID",
+					key: "dao_id",
+					dataIndex: "dao_id",
+				},
+				{
+					title: "Name",
+					key: "name",
+					dataIndex: "name",
+					customFilterDropdown: true,
+					onFilter: (value: string, record: DaoTableData) => {
+						return record.name
+							.toString()
+							.toLowerCase()
+							.includes(value.toLowerCase());
+					},
+
+					onFilterDropdownVisibleChange: (visible: boolean) => {
+						if (visible && this.$refs.searchInput) {
+							setTimeout(() => {
+								(this.$refs.searchInput as any).focus();
+							}, 100);
+						}
+					},
+				},
+				{
+					title: "Token Name",
+					key: "token_name",
+					dataIndex: "token_name",
+				},
+				{
+					title: "Token ID",
+					key: "token_id",
+					dataIndex: "token_id",
+				},
+				{
+					title: "Link to DAO",
+					key: "link",
+					dataIndex: "link",
+				},
+				{
+					title: "Action",
+					key: "action",
+				},
+			],
+			walletStore: WalletStore(),
 		};
 	},
 	methods: {
-		onFinish() {
+		handleSelectDAO(data: DaoTableData) {
+			this.formState.govt_id = data.token_id;
+			this.formState.dao_id = data.dao_id;
 			loadingMessage(this.key);
 			searchApplicationAndAccount()
 				.then(() => {
 					successMessage(this.key);
 					openSuccessNotificationWithIcon(
 						SUCCESSFUL,
-						daoAppMessage.SUCCESSFUL(this.formState.dao_id as number)
+						daoAppMessage.SUCCESSFUL(data.dao_id)
 					);
+					if (this.walletStore.address) {
+						isApplicationOpted(this.walletStore.address, data.dao_id)
+							.then((appIsOptedIn: boolean) => {
+								this.formState.show_opt_in = !appIsOptedIn;
+								if (appIsOptedIn) {
+									openSuccessNotificationWithIcon(daoAppMessage.ALREADY_OPT_IN);
+								}
+							})
+							.catch((error) =>
+								openErrorNotificationWithIcon(
+									daoAppMessage.UNSUCCESFUL,
+									error.message
+								)
+							);
+					}
 				})
 				.catch((error) => {
 					errorMessage(this.key);
-					this.error = error.message;
+					openErrorNotificationWithIcon(UNSUCCESSFUL, error.message);
 				});
 		},
-		onFinishFailed(errorinfo: Event) {
-			console.warn("Failed:", errorinfo);
+		handleDeSelectDao() {
+			this.formState.resetDaoStore();
 		},
 	},
 	setup() {
 		const formState = reactive(DaoStore());
+		const state = reactive({
+			searchText: "",
+			searchedColumn: "",
+		});
+		const searchInput = ref();
+		const tempArray: Array<DaoTableData> = [];
+		formState.psqlData = tempArray; // prohibit duplication of data
```

Comment:
We can move this inside at line 243. First push all in temp array and then make psqlData equal to temp array. In this way, if API fails, we will retain the existing data.

Reply:
Will update it in pagination, since there we will need to clean the data with page change

Reply:
Ok